src/components/NavBar/NavBarContainer.js

```diff
@@ -46,22 +46,22 @@ class NavBarContainer extends React.Component {
                 from NavBar.js, the context switches back to NavBarContainer.js.
             */
             <NavBar
-                authState={this.props.authState}
+                permissions={this.props.permissions}
                 navigate={this.navigate}
                 logout={this.logout}/>
         );
     }
 }
 
-// Used by mapStateToProps to get authState from the redux store
-const getAuthState = (authState) => {
-    return authState;
+// Used by mapStateToProps to get permissions from the redux store
+const getPermissions = (permissions) => {
```

Comment:
When `getPermissions` is passed to `mapStateToProps` it returns the `permissions` portion of the Redux store. `mapStateToProps` then maps `permissions` to RequireAuth's props through `this.props.permissions`. If you're asking why I don't define it elsewhere and import it here, I would say that you have a point.